modules/javafx.controls/src/main/java/javafx/scene/control/DialogPane.java

```diff
@@ -818,16 +817,15 @@ protected Node createDetailsButton() {
         final Hyperlink detailsButton = new Hyperlink();
         final String moreText = ControlResources.getString("Dialog.detail.button.more"); //$NON-NLS-1$
         final String lessText = ControlResources.getString("Dialog.detail.button.less"); //$NON-NLS-1$
+        final ObservableList<String> styleClasses = detailsButton.getStyleClass();
 
-        InvalidationListener expandedListener = o -> {
-            final boolean isExpanded = isExpanded();
-            detailsButton.setText(isExpanded ? lessText : moreText);
-            detailsButton.getStyleClass().setAll("details-button", (isExpanded ? "less" : "more")); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-        };
+        styleClasses.add("details-button");  //$NON-NLS-1$
```

Comment:
do we use these `//$NON-NLS-1$` in jfx?

Reply:
I do not know if we do, as there is also non-public code that I don't have access to.  I've kept them as-is to make sure the change is focused and doesn't break anything else.

If I had to guess then probably not, if you can ask around and find out I'll be happy to remove them.

Reply:
I'll double check and will let you know tomorrow.

Reply:
I can confirm that we don't use these `//$NON-NLS-*` comments in javaFX.

`<rant>`
These comments were invented as a way to solve the localization problem.  The problem is real, but the "solution" was exactly the opposite.  Instead of marking the strings that don't need to be localized, what should have happened is to mark the string that need to be localized - with the information containing the context!

example:
```java
// verb, as in "execute this test"
String run = "Run";
```

What I've done with some other projects was 

```java
String run = TXT.get("Class.LocalizedResourceID.verb, as in execute this test", "Run");
```

`</rant>`

Reply:
Okay, I can remove them.

Reply:
line 822 should be

`styleClasses.setAll("details-button");`